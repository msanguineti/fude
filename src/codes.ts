/**
 * 文字装飾コード番号 - The character OrnamentCode (SGR Parameter) to apply
 *
 * @see https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters
 */

export type OrnamentCode = number

/** @constant reset. All attributes off.*/
export const resetCode: OrnamentCode = 0

/** @constant bold. As with faint/dim, the color change is a PC (SCO / CGA) invention.*/
export const boldCode: OrnamentCode = 1

/** @constant dim  (with a saturated color). May be implemented as a light font weight like bold.*/
export const dimCode: OrnamentCode = 2

/** @constant italic. Not widely supported.. Sometimes treated as inverse or blink.*/
export const italicCode: OrnamentCode = 3

/** @constant underline.. Style extensions exist for Kitty, VTE, mintty and iTerm2.*/
export const underlineCode: OrnamentCode = 4

/** @constant blinkSlow; less than 150 per minute.*/
export const blinkSlowCode: OrnamentCode = 5

/** @constant blinkRapid. MS-DOS ANSI.SYS, 150+ per minute; not widely supported.*/
export const blinkRapidCode: OrnamentCode = 6

/** @constant invert; swap foreground and background colors; inconsistent emulation.*/
export const invertCode: OrnamentCode = 7

/** @constant hide; not widely supported.*/
export const hideCode: OrnamentCode = 8

/** @constant strikethrough. Characters legible but marked as if for deletion.*/
export const strikethroughCode: OrnamentCode = 9

/** @constant defaultFont. Primary (default) font.*/
export const defaultFontCode: OrnamentCode = 10

/** @constant altFont1Code: alternative font */
export const altFont1Code: OrnamentCode = 11

/** @constant altFont2Code: alternative font */
export const altFont2Code: OrnamentCode = 12

/** @constant altFont3Code: alternative font */
export const altFont3Code: OrnamentCode = 13

/** @constant altFont4Code: alternative font */
export const altFont4Code: OrnamentCode = 14

/** @constant altFont5Code: alternative font */
export const altFont5Code: OrnamentCode = 15

/** @constant altFont6Code: alternative font */
export const altFont6Code: OrnamentCode = 16

/** @constant altFont7Code: alternative font */
export const altFont7Code: OrnamentCode = 17

/** @constant altFont8Code: alternative font */
export const altFont8Code: OrnamentCode = 18

/** @constant altFont9Code: alternative font */
export const altFont9Code: OrnamentCode = 19

/** @constant fraktur. Rarely supported.*/
export const frakturCode: OrnamentCode = 20

/** @constant doublyUnderline. Doubly underline or Bold off. Double-underline per ECMA-48.:8.3.117. See discussion.*/
export const doublyUnderlineCode: OrnamentCode = 21

/** @constant normalColor. Normal color or intensity. Neither bold nor faint/ dim.*/
export const normalColorCode: OrnamentCode = 22

/** @constant resetItalic Not italic, not Fraktur.*/
export const resetItalicCode: OrnamentCode = 23

/** @constant resetUnderline. Underline off. Not singly or doubly underlined.*/
export const resetUnderlineCode: OrnamentCode = 24

/** @constant resetBlink. Blink off.*/
export const resetBlinkCode: OrnamentCode = 25

/** @constant propSpacing Proportional spacing. ITU T.61 and T.416, not known to be used on terminals.*/
export const propSpacingCode: OrnamentCode = 26

/** @constant resetInverse Reverse/invert off.*/
export const resetInverseCode: OrnamentCode = 27

/** @constant resetHide Reveal; conceal/hide off.*/
export const resetHideCode: OrnamentCode = 28

/** @constant resetStrikethrough Not crossed out.*/
export const resetStrikethroughCode: OrnamentCode = 29

/** @constant black. Set foreground color.*/
export const blackCode: OrnamentCode = 30

/** @constant red. Set foreground color.*/
export const redCode: OrnamentCode = 31

/** @constant green. Set foreground color.*/
export const greenCode: OrnamentCode = 32

/** @constant yellow. Set foreground color.*/
export const yellowCode: OrnamentCode = 33

/** @constant blue. Set foreground color.*/
export const blueCode: OrnamentCode = 34

/** @constant magenta. Set foreground color.*/
export const magentaCode: OrnamentCode = 35

/** @constant cyan. Set foreground color.*/
export const cyanCode: OrnamentCode = 36

/** @constant white. Set foreground color.*/
export const whiteCode: OrnamentCode = 37

/** @constant resetForeground Default foreground color; implementation defined (according to standard).*/
export const resetForegroundCode: OrnamentCode = 39

/** @constant bgBlack. Set background color.*/
export const bgBlackCode: OrnamentCode = 40

/** @constant bgRed. Set background color.*/
export const bgRedCode: OrnamentCode = 41

/** @constant bgGreen. Set background color.*/
export const bgGreenCode: OrnamentCode = 42

/** @constant bgYellow. Set background color.*/
export const bgYellowCode: OrnamentCode = 43

/** @constant bgBlue. Set background color.*/
export const bgBlueCode: OrnamentCode = 44

/** @constant bgMagenta. Set background color.*/
export const bgMagentaCode: OrnamentCode = 45

/** @constant bgCyan. Set background color.*/
export const bgCyanCode: OrnamentCode = 46

/** @constant bgWhite. Set background color.*/
export const bgWhiteCode: OrnamentCode = 47

/** @constant bgReset Default background color; implementation defined (according to standard).*/
export const resetBackgroundCode: OrnamentCode = 49

/** @constant resetPropSpacing Disable proportional spacing. T.61 and T.416*/
export const resetPropSpacingCode: OrnamentCode = 50

/** @constant framed. Implemented as "emoji variation selector" in mintty.*/
export const framedCode: OrnamentCode = 51

/** @constant encircled Implemented as "emoji variation selector" in mintty.*/
export const encircledCode: OrnamentCode = 52

/** @constant overlined Implemented as "emoji variation selector" in mintty.*/
export const overlinedCode: OrnamentCode = 53

/** @constant resetFramed. Not framed or encircled.*/
export const resetFramedCode: OrnamentCode = 54

/** @constant resetOverlines Not overlined.*/
export const resetOverlinesCode: OrnamentCode = 55

/** @constant ideogramU ideogram underline or right side line. Rarely supported.*/
export const ideogramUCode: OrnamentCode = 60

/** @constant ideogramDU ideogram double underline on; double line on the right side.*/
export const ideogramDUCode: OrnamentCode = 61

/** @constant ideogramO ideogram overline or left side line.*/
export const ideogramOCode: OrnamentCode = 62

/** @constant ideogramDO ideogram double overline on; double line on the left side.*/
export const ideogramDOCode: OrnamentCode = 63

/** @constant ideogramS ideogram stress marking.*/
export const ideogramSCode: OrnamentCode = 64

/** @constant ideogramOff ideogram attributes off; reset the effects of all of 60–64*/
export const ideogramOffCode: OrnamentCode = 65

/** @constant superscript mintty (not in standard).*/
export const superscriptCode: OrnamentCode = 73

/** @constant subscript mintty (not in standard).*/
export const subscriptCode: OrnamentCode = 74

/** @constant gray.. Set bright foreground color; aixterm (not in standard).*/
export const grayCode: OrnamentCode = 90

/** @constant brightRed. Set bright foreground color; aixterm (not in standard).*/
export const brightRedCode: OrnamentCode = 91

/** @constant brightGreen. Set bright foreground color; aixterm (not in standard).*/
export const brightGreenCode: OrnamentCode = 92

/** @constant brightYellow. Set bright foreground color; aixterm (not in standard).*/
export const brightYellowCode: OrnamentCode = 93

/** @constant brightBlue. Set bright foreground color; aixterm (not in standard).*/
export const brightBlueCode: OrnamentCode = 94

/** @constant brightMagenta. Set bright foreground color; aixterm (not in standard).*/
export const brightMagentaCode: OrnamentCode = 95

/** @constant brightCyan. Set bright foreground color; aixterm (not in standard).*/
export const brightCyanCode: OrnamentCode = 96

/** @constant brightWhite. Set bright foreground color; aixterm (not in standard).*/
export const brightWhiteCode: OrnamentCode = 97

/** @constant bgBrightBlack. Set bright background color.*/
export const bgBrightBlackCode: OrnamentCode = 100

/** @constant bgBrightRed. Set bright background color.*/
export const bgBrightRedCode: OrnamentCode = 101

/** @constant bgBrightGreen. Set bright background color.*/
export const bgBrightGreenCode: OrnamentCode = 102

/** @constant bgBrightYellow. Set bright background color.*/
export const bgBrightYellowCode: OrnamentCode = 103

/** @constant bgBrightBlue. Set bright background color.*/
export const bgBrightBlueCode: OrnamentCode = 104

/** @constant bgBrightMagenta. Set bright background color.*/
export const bgBrightMagentaCode: OrnamentCode = 105

/** @constant bgBrightCyan. Set bright background color.*/
export const bgBrightCyanCode: OrnamentCode = 106

/** @constant bgBrightWhite. Set bright background color.*/
export const bgBrightWhiteCode: OrnamentCode = 107

/**
 * Prints what is available on your TTY
 */
export const ttyCapability = (): string => {
  const output = []

  for (let i = 0; i < 109; i++) {
    const text = `\x1b[${i}m${(i + '').padStart(3)}\x1b[0m${
      (i + 1) % 10 === 0 ? '\n' : ' '
    }`
    output.push(text)
  }

  return output.join('')
}
