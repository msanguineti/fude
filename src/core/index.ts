import { CharacterOrnament } from '../ornaments'

/**
 * 書く - composes the various text ornaments to obtain the final output string
 *
 * @param tos set of text ornaments
 * @param text the input string
 * @returns a string with the applied text ornaments
 */
const kaku = (tos: CharacterOrnament[]) => (text: string) =>
  tos.reduceRight((v, f) => f(v), text)

/**
 * 筆 - Takes a string and applies the specified text ornaments to it.
 *
 * @param text the input string
 * @param tos text ornaments to apply
 */
export const fude = (text: string, ...tos: CharacterOrnament[]): string =>
  kaku(tos)(text)
