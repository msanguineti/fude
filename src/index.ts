export * from './core'
export * from './ornaments'
export * from './codes'
