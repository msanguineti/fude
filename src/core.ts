import { OrnamentCode } from './codes'
import { CharacterOrnament } from './ornaments'

/**
 * This code was taken from Kleur (https://github.com/lukeed/kleur) and adapted for our use case.
 *
 * FORCE_COLOR should decide whether to force colors or not.
 */
const { FORCE_COLOR, NODE_DISABLE_COLORS, NO_COLOR, TERM } = process.env
const isTTY = process.stdout && process.stdout.isTTY
let enabled = FORCE_COLOR
  ? !/^\s*(?:false|0|off)\s*$/i.test(FORCE_COLOR)
  : !NODE_DISABLE_COLORS && NO_COLOR == null && TERM !== 'dumb' && isTTY

export const setEnabled = (colorsEnabled: boolean): void => {
  enabled = colorsEnabled
}

export const isEnabled = (): boolean => enabled

const templates = (txt, keys, opt) => {
  let ret = txt[0]
  for (let i = 0; i < keys.length; i++) {
    ret += keys[i] + txt[i + 1]
  }

  return enabled
    ? opt.open +
        (ret.indexOf(opt.close) >= 0 ? ret.replace(opt.rgx, opt.open) : ret) +
        opt.close
    : ret
}
/**
 * 筆致 - Only apply the given attribute to the input string if we are in a TTY environment or if the FORCE_COLOR flag is set.
 *
 * @param text the input string
 * @param on the attribute on code
 * @param off the attribute off code
 * @returns a string with the applied attribute
 */
export const hitchi = (
  on: OrnamentCode,
  off: OrnamentCode = 0
): ((...text: string[]) => string) => {
  const opt = {
    open: `\x1b[${on}m`,
    close: `\x1b[${off}m`,
    rgx: new RegExp(`\\x1b\\[(?:${off})m`, 'gi'),
  }
  return (txt, ...keys): string =>
    keys.length !== 0
      ? templates(txt, keys, opt)
      : enabled
      ? opt.open +
        (txt.indexOf(opt.close) >= 0 ? txt.replace(opt.rgx, opt.open) : txt) +
        opt.close
      : txt + ''
}

/**
 * 筆 - Takes a string and applies the specified text ornaments to it.
 *
 * @param text the input string
 * @param tos text ornaments to apply
 */
export const fude = (
  text: string | CharacterOrnament,
  ...tos: CharacterOrnament[]
): string => tos.reduceRight((v, f) => f(v), text as string)

// match close attributes escape codes at the end of a given string
const rgx = new RegExp(`(?:\\x1b\\[[0-9]{1,3}m)+$`, 'i')

const closeAttribute = {
  0: 0,
  1: 22,
  2: 22,
  3: 23,
  4: 24,
  21: 24,
  5: 25,
  6: 26,
  7: 27,
  8: 28,
  9: 29,
  11: 10,
  12: 10,
  13: 10,
  14: 10,
  15: 10,
  16: 10,
  17: 10,
  18: 10,
  19: 10,
  20: 23,
  26: 50,
  30: 39,
  31: 39,
  32: 39,
  33: 39,
  34: 39,
  35: 39,
  36: 39,
  37: 39,
  38: 39,
  40: 49,
  41: 49,
  42: 49,
  43: 49,
  44: 49,
  45: 49,
  46: 49,
  47: 49,
  48: 49,
  51: 54,
  52: 54,
  53: 55,
  58: 59,
  60: 65,
  61: 65,
  62: 65,
  63: 65,
  64: 65,
  73: 10,
  74: 10,
  90: 39,
  91: 39,
  92: 39,
  93: 39,
  94: 39,
  95: 39,
  96: 39,
  97: 39,
  100: 49,
  101: 49,
  102: 49,
  103: 49,
  104: 49,
  105: 49,
  106: 49,
  107: 49,
}

/**
 * Pass ANSI SGR attributes codes directly
 *
 * @param text the input string
 * @param codes the OrnamentCode codes to apply to the input
 */
export const ansi = (
  text: string | CharacterOrnament,
  ...codes: OrnamentCode[]
): string =>
  enabled
    ? `\x1b[${codes.join(';')}m` +
      ((text as string).indexOf('\x1b[') >= 0
        ? (text as string).replace(rgx, '')
        : text) +
      `\x1b[${codes.map((c) => closeAttribute[c] ?? 0)}m`
    : (text as string)

const applyRGB = (
  r: number,
  g: number,
  b: number,
  text: string | CharacterOrnament,
  open,
  close
): string =>
  hitchi(
    (`${open};${2};${r};${g};${b}` as unknown) as number,
    close
  )(text as string)

const openRGBColorsAttribute = 38
const closeRGBColorsAttribute = 39

/**
 * Apply an RGB foreground color to the given input text.
 *
 * @param text the input string
 * @param r red color value (0-255) default 0
 * @param g green color value (0-255) default 0
 * @param b blue color attribute (0-255) default 0
 */
export const rgb = (
  text: string | CharacterOrnament,
  { r = 0, g = 0, b = 0 }: { r?: number; g?: number; b?: number }
): string =>
  enabled
    ? applyRGB(r, g, b, text, openRGBColorsAttribute, closeRGBColorsAttribute)
    : (text as string)

const openRGBBGColorsAttribute = 48
const closeRGBBGColorsAttribute = 49

/**
 * Apply an RGB background color to the given input text.
 *
 * @param text the input string
 * @param r red color value (0-255) default 0
 * @param g green color value (0-255) default 0
 * @param b blue color attribute (0-255) default 0
 */
export const rgbBg = (
  text: string | CharacterOrnament,
  { r = 0, g = 0, b = 0 }: { r?: number; g?: number; b?: number }
): string =>
  enabled
    ? applyRGB(
        r,
        g,
        b,
        text,
        openRGBBGColorsAttribute,
        closeRGBBGColorsAttribute
      )
    : (text as string)

const openRGBUnderlineColorsAttribute = 58
const closeRGBUnderlineColorsAttribute = 59

/**
 * Apply an RGB color to the underline of the given input text.
 *
 * @param text the input string
 * @param r red color value (0-255) default 0
 * @param g green color value (0-255) default 0
 * @param b blue color attribute (0-255) default 0
 * @param double doubly underline default to `false`
 */
export const rgbUnderline = (
  text: string | CharacterOrnament,
  {
    r = 0,
    g = 0,
    b = 0,
    double = false,
  }: { r?: number; g?: number; b?: number; double?: boolean }
): string =>
  enabled
    ? applyRGB(
        r,
        g,
        b,
        double ? hitchi(21, 24)(text as string) : hitchi(4, 24)(text as string),
        openRGBUnderlineColorsAttribute,
        closeRGBUnderlineColorsAttribute
      )
    : (text as string)

/**
 * Apply a HEX foreground color value to the given input text.
 *
 * Case insensitive, with or without `#` as a prefix.
 *
 * @param text the input string
 * @param hex the color value as an hex number
 */
export const hex = (text: string | CharacterOrnament, hex: string): string => {
  const match = /([0-9a-f]{2})([0-9a-f]{2})([0-9a-f]{2})/i.exec(hex)

  return match
    ? rgb(text, {
        r: parseInt(match[1], 16),
        g: parseInt(match[2], 16),
        b: parseInt(match[3], 16),
      })
    : (text as string)
}

/**
 * Apply a HEX background color value to the given input text.
 *
 * Case insensitive, with or without `#` as a prefix.
 *
 * @param text the input string
 * @param hex the color value as an hex number
 */
export const hexBg = (
  text: string | CharacterOrnament,
  hex: string
): string => {
  const match = /([0-9a-f]{2})([0-9a-f]{2})([0-9a-f]{2})/i.exec(hex)

  return match
    ? rgbBg(text, {
        r: parseInt(match[1], 16),
        g: parseInt(match[2], 16),
        b: parseInt(match[3], 16),
      })
    : (text as string)
}

/**
 * Apply a HEX color value to the underline of the given input text.
 *
 * Case insensitive, with or without `#` as a prefix.
 *
 * @param text the input string
 * @param hex the color value as an hex number
 */
export const hexUnderline = (
  text: string | CharacterOrnament,
  hex: string,
  double = false
): string => {
  const match = /([0-9a-f]{2})([0-9a-f]{2})([0-9a-f]{2})/i.exec(hex)

  return match
    ? rgbUnderline(text, {
        r: parseInt(match[1], 16),
        g: parseInt(match[2], 16),
        b: parseInt(match[3], 16),
        double,
      })
    : (text as string)
}
