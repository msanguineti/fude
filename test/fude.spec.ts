import * as fude from '../src'

describe('core functionalities', () => {
  test('single ornament', () => {
    const output = { text: fude.fude('bar', fude.red) }
    expect(output).toMatchSnapshot({
      text: expect.stringContaining('\x1b[31mbar\x1b[39m'),
    })
  })

  test('two ornaments', () => {
    const output = { text: fude.fude('bar', fude.red, fude.bold) }
    expect(output).toMatchSnapshot({
      text: expect.stringContaining('\x1b[31m\x1b[1mbar\x1b[22m\x1b[39m'),
    })
  })

  test('colours are switched off', () => {
    process.env.FORCE_COLOURS = '0'
    const output = { text: fude.fude('bar', fude.red) }
    expect(output).toMatchSnapshot({ text: expect.stringMatching('bar') })
    process.env.FORCE_COLOURS = '1'
  })
})

describe('ornaments', () => {
  test('modifiers', () => {
    const output = {
      text: fude.reset(
        `reset ${fude.bold('bold')} ${fude.dim('dim')} ${fude.italic(
          'italic'
        )} ${fude.underline('underline')} ${fude.doublyUnderline(
          'doublyUnderline'
        )} ${fude.blinkSlow('slow')} ${fude.blinkFast('fast')} ${fude.invert(
          'invert'
        )} ${fude.hide('hide')} ${fude.strikethrough('strikethrough')}`
      ),
    }
    expect(output).toMatchSnapshot({
      text: expect.stringContaining(
        '\x1b[0mreset \x1b[1mbold\x1b[22m \x1b[2mdim\x1b[22m \x1b[3mitalic\x1b[23m \x1b[4munderline\x1b[24m \x1b[21mdoublyUnderline\x1b[24m \x1b[5mslow\x1b[25m \x1b[6mfast\x1b[26m \x1b[7minvert\x1b[27m \x1b[8mhide\x1b[28m \x1b[9mstrikethrough\x1b[29m\x1b[0m'
      ),
    })
  })

  test('foreground colours', () => {
    const output = {
      text: fude.black(
        `black ${fude.red('red')} ${fude.green('green')} ${fude.yellow(
          'yellow'
        )} ${fude.blue('blue')} ${fude.magenta('magenta')} ${fude.cyan(
          'cyan'
        )} ${fude.white('white')} ${fude.gray('gray')}`
      ),
    }
    expect(output).toMatchSnapshot({
      text: expect.stringContaining(
        '\x1b[30mblack \x1b[31mred\x1b[39m \x1b[32mgreen\x1b[39m \x1b[33myellow\x1b[39m \x1b[34mblue\x1b[39m \x1b[35mmagenta\x1b[39m \x1b[36mcyan\x1b[39m \x1b[37mwhite\x1b[39m \x1b[90mgray\x1b[39m\x1b[39'
      ),
    })
  })

  test('background colours', () => {
    const output = {
      text: fude.bgBlack(
        `black ${fude.bgRed('red')} ${fude.bgGreen('green')} ${fude.bgYellow(
          'yellow'
        )} ${fude.bgBlue('blue')} ${fude.bgMagenta('magenta')} ${fude.bgCyan(
          'cyan'
        )} ${fude.bgWhite('white')} ${fude.bgGray('gray')}`
      ),
    }
    expect(output).toMatchSnapshot(
      '\x1b[40mblack \x1b[41mred\x1b[49m \x1b[42mgreen\x1b[49m \x1b[43myellow\x1b[49m \x1b[44mblue\x1b[49m \x1b[45mmagenta\x1b[49m \x1b[46mcyan\x1b[49m \x1b[47mwhite\x1b[49m \x1b[100mgray\x1b[49m\x1b[49m'
    )
  })

  test('bright foreground colours', () => {
    const output = {
      text: fude.gray(
        `gray ${fude.brightRed('red')} ${fude.brightGreen(
          'green'
        )} ${fude.brightYellow('yellow')} ${fude.brightBlue(
          'blue'
        )} ${fude.brightMagenta('magenta')} ${fude.brightCyan(
          'cyan'
        )} ${fude.brightWhite('white')}`
      ),
    }
    expect(output).toMatchSnapshot(
      '\x1b[90mblack \x1b[91mred\x1b[39m \x1b[92mgreen\x1b[39m \x1b[93myellow\x1b[39m \x1b[94mblue\x1b[39m \x1b[95mmagenta\x1b[39m \x1b[96mcyan\x1b[39m \x1b[97mwhite\x1b[39m\x1b[39m'
    )
  })

  test('bright background colours', () => {
    const output = {
      text: fude.bgGray(
        `gray ${fude.bgBrightRed('red')} ${fude.bgBrightGreen(
          'green'
        )} ${fude.bgBrightYellow('yellow')} ${fude.bgBrightBlue(
          'blue'
        )} ${fude.bgBrightMagenta('magenta')} ${fude.bgBrightCyan(
          'cyan'
        )} ${fude.bgBrightWhite('white')} `
      ),
    }
    expect(output).toMatchSnapshot(
      '\x1b[100mblack \x1b[101mred\x1b[49m \x1b[102mgreen\x1b[49m \x1b[103myellow\x1b[49m \x1b[104mblue\x1b[49m \x1b[105mmagenta\x1b[49m \x1b[106mcyan\x1b[49m \x1b[107mwhite\x1b[49m\x1b[49m'
    )
  })

  test('using codes', () => {
    const output = { text: fude.fudeCodes('foo', fude.yellowCode) }
    expect(output).toMatchSnapshot({
      text: expect.stringContaining('\x1b[33mfoo\x1b[0m'),
    })
  })
})

describe('util', () => {
  test('all the codes', () => {
    const output = { text: fude.availableOrnamentCodes() }
    expect(output).toMatchSnapshot()
  })
})
