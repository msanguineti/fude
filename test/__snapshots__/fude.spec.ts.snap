// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core functionalities colours are switched off 1`] = `
Object {
  "text": StringMatching /bar/,
}
`;

exports[`core functionalities single ornament 1`] = `
Object {
  "text": StringContaining "[31mbar[39m",
}
`;

exports[`core functionalities two ornaments 1`] = `
Object {
  "text": StringContaining "[31m[1mbar[22m[39m",
}
`;

exports[`ornaments background colours: [40mblack [41mred[49m [42mgreen[49m [43myellow[49m [44mblue[49m [45mmagenta[49m [46mcyan[49m [47mwhite[49m [100mgray[49m[49m 1`] = `
Object {
  "text": "[40mblack [41mred[49m [42mgreen[49m [43myellow[49m [44mblue[49m [45mmagenta[49m [46mcyan[49m [47mwhite[49m [100mgray[49m[49m",
}
`;

exports[`ornaments bright background colours: [100mblack [101mred[49m [102mgreen[49m [103myellow[49m [104mblue[49m [105mmagenta[49m [106mcyan[49m [107mwhite[49m[49m 1`] = `
Object {
  "text": "[100mgray [101mred[49m [102mgreen[49m [103myellow[49m [104mblue[49m [105mmagenta[49m [106mcyan[49m [107mwhite[49m [49m",
}
`;

exports[`ornaments bright foreground colours: [90mblack [91mred[39m [92mgreen[39m [93myellow[39m [94mblue[39m [95mmagenta[39m [96mcyan[39m [97mwhite[39m[39m 1`] = `
Object {
  "text": "[90mgray [91mred[39m [92mgreen[39m [93myellow[39m [94mblue[39m [95mmagenta[39m [96mcyan[39m [97mwhite[39m[39m",
}
`;

exports[`ornaments foreground colours 1`] = `
Object {
  "text": StringContaining "[30mblack [31mred[39m [32mgreen[39m [33myellow[39m [34mblue[39m [35mmagenta[39m [36mcyan[39m [37mwhite[39m [90mgray[39m[39",
}
`;

exports[`ornaments modifiers 1`] = `
Object {
  "text": StringContaining "[0mreset [1mbold[22m [2mdim[22m [3mitalic[23m [4munderline[24m [21mdoublyUnderline[24m [5mslow[25m [6mfast[26m [7minvert[27m [8mhide[28m [9mstrikethrough[29m[0m",
}
`;

exports[`ornaments using codes 1`] = `
Object {
  "text": StringContaining "[33mfoo[0m",
}
`;

exports[`util all the codes 1`] = `
Object {
  "text": "[0m  0[0m [1m  1[0m [2m  2[0m [3m  3[0m [4m  4[0m [5m  5[0m [6m  6[0m [7m  7[0m [8m  8[0m [9m  9[0m
[10m 10[0m [11m 11[0m [12m 12[0m [13m 13[0m [14m 14[0m [15m 15[0m [16m 16[0m [17m 17[0m [18m 18[0m [19m 19[0m
[20m 20[0m [21m 21[0m [22m 22[0m [23m 23[0m [24m 24[0m [25m 25[0m [26m 26[0m [27m 27[0m [28m 28[0m [29m 29[0m
[30m 30[0m [31m 31[0m [32m 32[0m [33m 33[0m [34m 34[0m [35m 35[0m [36m 36[0m [37m 37[0m [38m 38[0m [39m 39[0m
[40m 40[0m [41m 41[0m [42m 42[0m [43m 43[0m [44m 44[0m [45m 45[0m [46m 46[0m [47m 47[0m [48m 48[0m [49m 49[0m
[50m 50[0m [51m 51[0m [52m 52[0m [53m 53[0m [54m 54[0m [55m 55[0m [56m 56[0m [57m 57[0m [58m 58[0m [59m 59[0m
[60m 60[0m [61m 61[0m [62m 62[0m [63m 63[0m [64m 64[0m [65m 65[0m [66m 66[0m [67m 67[0m [68m 68[0m [69m 69[0m
[70m 70[0m [71m 71[0m [72m 72[0m [73m 73[0m [74m 74[0m [75m 75[0m [76m 76[0m [77m 77[0m [78m 78[0m [79m 79[0m
[80m 80[0m [81m 81[0m [82m 82[0m [83m 83[0m [84m 84[0m [85m 85[0m [86m 86[0m [87m 87[0m [88m 88[0m [89m 89[0m
[90m 90[0m [91m 91[0m [92m 92[0m [93m 93[0m [94m 94[0m [95m 95[0m [96m 96[0m [97m 97[0m [98m 98[0m [99m 99[0m
[100m100[0m [101m101[0m [102m102[0m [103m103[0m [104m104[0m [105m105[0m [106m106[0m [107m107[0m [108m108[0m ",
}
`;
